<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>identify_markers computes differential observation expression 
between selected territories. — identify_markers • vesalius</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="identify_markers computes differential observation expression 
between selected territories. — identify_markers"><meta property="og:description" content="identify_markers computes differential observation expression 
between selected territories."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">vesalius</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/vesalius.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Vesalius_Analysis/Spatial_DNA_seq.html">Vesalius: Spatial Genomics</a>
    </li>
    <li>
      <a href="../articles/Vesalius_Analysis/Spatial_Transcriptomics.html">Vesalius: Spatial Transcriptomics</a>
    </li>
    <li>
      <a href="../articles/Vesalius_Analysis/Vesalius_MSB_analysis.html">Vesalius: MSB Manuscript Analysis</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/patrickCNMartin/Vesalius/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>identify_markers computes differential observation expression 
between selected territories.</h1>
    <small class="dont-index">Source: <a href="https://github.com/patrickCNMartin/Vesalius/blob/main/R/territory_identity_and_markers.R" class="external-link"><code>R/territory_identity_and_markers.R</code></a></small>
    <div class="hidden name"><code>identify_markers.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>identify_markers computes differential observation expression 
between selected territories.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">identify_markers</span><span class="op">(</span></span>
<span>  <span class="va">vesalius_assay</span>,</span>
<span>  trial <span class="op">=</span> <span class="st">"last"</span>,</span>
<span>  norm_method <span class="op">=</span> <span class="st">"last"</span>,</span>
<span>  seed <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  query <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cells <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"wilcox"</span>,</span>
<span>  log_fc <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  pval <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  min_pct <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  min_spatial_index <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>vesalius_assay</dt>
<dd><p>a vesalius_assay</p></dd>


<dt>trial</dt>
<dd><p>character string - which territory trial that 
should be used to select
territorires. Default is last one computed</p></dd>


<dt>norm_method</dt>
<dd><p>charcater string - which normalisation method should 
be used.</p></dd>


<dt>seed</dt>
<dd><p>Integer or vector of integers describing territories to be
included in group 1 for differential gene expression analysis.</p></dd>


<dt>query</dt>
<dd><p>Integer or vector of integers describing territories to be
included in group 2 for differential gene expression analysis. Default = NULL</p></dd>


<dt>cells</dt>
<dd><p>character vector containing barcodes of cells of interest.</p></dd>


<dt>method</dt>
<dd><p>character describing the statistical test to use in order to
extract differantial gene expression.
Select from:
"wilcox", "t.test", "chisq", "fisher.exact", "DEseq2", "QLF", "LRT","logit"</p></dd>


<dt>log_fc</dt>
<dd><p>numeric describing minimum log fold change value for
differential gene expression. Default set at 0.25.</p></dd>


<dt>pval</dt>
<dd><p>numeric for pval threshold. Default set at 0.05</p></dd>


<dt>min_pct</dt>
<dd><p>numeric defining the minimum percentage of cells that should
contain any given gene. Deault set at 0.05</p></dd>


<dt>min_spatial_index</dt>
<dd><p>integer defining minimum number of 
barcodes in a territory.</p></dd>


<dt>verbose</dt>
<dd><p>logical - progress message output</p></dd>


<dt>...</dt>
<dd><p>other parameters parsed to DESeq2 or edgeR (not functional)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a vesalius_assay object</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Identifying markers is a key aspect of spatial data analysis. 
This functions let's you select which territory trial you which to use.
Note that this can be any territory trial that you have run, including 
color segments, isolated territories, dilated or eroded territories and
layered territories. By default, <code>identify_markers</code> takes the last 
one that has been computed.</p>
<p>The normalisation method refers to the normalisation method applied to
the count matrices. If you use, DESeq2, QLF (edgeR) or LRT (edgeR), 
raw counts will be selected and will ignore any cother command. 
This is a requirement for both of these packages.</p>
<p>If you have some cells you are interested in comparing between territories,
you can simply parse a character vector containing the barcodes of your 
cells of interest. <code>identify_markers</code> will automatically retrieve cells
in each territory and only use these cell for comparison.</p>
<p>Once the Differentially expressed genes/oberservations have been computed 
they are stored in the vesalius_assay object. This allows you to run multiple
trial and have all these trials sorted within your object.</p>
<p>To retrieve them from the object, you can use <code><a href="get_markers.html">get_markers</a></code></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">vesalius</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># First we build a simple object</span></span></span>
<span class="r-in"><span><span class="va">ves</span> <span class="op">&lt;-</span> <span class="fu">build_vesalius_object</span><span class="op">(</span><span class="va">coordinates</span>, <span class="va">counts</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># We can do a simple run</span></span></span>
<span class="r-in"><span><span class="va">ves</span> <span class="op">&lt;-</span> <span class="fu">build_vesalius_embeddings</span><span class="op">(</span><span class="va">ves</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># simple smoothing</span></span></span>
<span class="r-in"><span><span class="va">ves</span> <span class="op">&lt;-</span> <span class="fu"><a href="smooth_image.html">smooth_image</a></span><span class="op">(</span><span class="va">ves</span>, dimensions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">30</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># quick segmentation</span></span></span>
<span class="r-in"><span><span class="va">ves</span> <span class="op">&lt;-</span> <span class="fu"><a href="segment_image.html">segment_image</a></span><span class="op">(</span><span class="va">ves</span>, dimensions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">30</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># isolate territories</span></span></span>
<span class="r-in"><span><span class="va">ves</span> <span class="op">&lt;-</span> <span class="fu"><a href="isolate_territories.html">isolate_territories</a></span><span class="op">(</span><span class="va">ves</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># identify markers</span></span></span>
<span class="r-in"><span><span class="va">ves</span> <span class="op">&lt;-</span> <span class="fu">identify_markers</span><span class="op">(</span><span class="va">ves</span>, seed <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">5</span><span class="op">)</span>, query <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="get_markers.html">get_markers</a></span><span class="op">(</span><span class="va">ves</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Patrick C.N. Martin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

