<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ vesalius</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">vesalius</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/vesalius.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Vesalius_Analysis/Spatial_DNA_seq.html">Vesalius: Spatial Genomics</a>
    </li>
    <li>
      <a href="../articles/Vesalius_Analysis/Spatial_Transcriptomics.html">Vesalius: Spatial Transcriptomics</a>
    </li>
    <li>
      <a href="../articles/Vesalius_Analysis/Vesalius_MSB_analysis.html">Vesalius: MSB Manuscript Analysis</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/patrickCNMartin/Vesalius/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="add_active_count_tag.html">add_active_count_tag()</a></code> </p>
        </td>
        <td><p>add active count tag</p></td>
      </tr><tr><td>
          <p><code><a href="add_active_embedding_tag.html">add_active_embedding_tag()</a></code> </p>
        </td>
        <td><p>add active embedding tag</p></td>
      </tr><tr><td>
          <p><code><a href="add_counts.html">add_counts()</a></code> </p>
        </td>
        <td><p>Add counts to vesalius assay
Adding custom count matrix to a vesalius assay object.</p></td>
      </tr><tr><td>
          <p><code><a href="add_embeddings.html">add_embeddings()</a></code> </p>
        </td>
        <td><p>Add embeddings
Add custom embeddings to vesalius objects</p></td>
      </tr><tr><td>
          <p><code><a href="adjust_counts.html">adjust_counts()</a></code> </p>
        </td>
        <td><p>adjust count</p></td>
      </tr><tr><td>
          <p><code><a href="align_index.html">align_index()</a></code> </p>
        </td>
        <td><p>assign coordinates to matched indices</p></td>
      </tr><tr><td>
          <p><code><a href="assign_centers.html">assign_centers()</a></code> </p>
        </td>
        <td><p>assign centroid values to active embedding</p></td>
      </tr><tr><td>
          <p><code><a href="average_embed.html">average_embed()</a></code> </p>
        </td>
        <td><p>average image stack between seed and query</p></td>
      </tr><tr><td>
          <p><code><a href="bubble_stack.html">bubble_stack()</a></code> </p>
        </td>
        <td><p>importFrom imager nPix</p></td>
      </tr><tr><td>
          <p><code><a href="build_vesalius_assay.html">build_vesalius_assay()</a></code> </p>
        </td>
        <td><p>build vesalius assay object</p></td>
      </tr><tr><td>
          <p><code><a href="calculate_scale.html">calculate_scale()</a></code> </p>
        </td>
        <td><p>calculate scale of assay</p></td>
      </tr><tr><td>
          <p><code><a href="check_barcodes.html">check_barcodes()</a></code> </p>
        </td>
        <td><p>checking overlap between barcodes in counts and coordinates</p></td>
      </tr><tr><td>
          <p><code><a href="check_binary_nature.html">check_binary_nature()</a></code> </p>
        </td>
        <td><p>check binary nature</p></td>
      </tr><tr><td>
          <p><code><a href="check_cells.html">check_cells()</a></code> </p>
        </td>
        <td><p>check if provided cells are contained withing provided territory</p></td>
      </tr><tr><td>
          <p><code><a href="check_coordinates.html">check_coordinates()</a></code> </p>
        </td>
        <td><p>check if coordinates are of the correct type and format and
adjust coordinate value to remove white edge space</p></td>
      </tr><tr><td>
          <p><code><a href="check_counts.html">check_counts()</a></code> </p>
        </td>
        <td><p>check if counts are of the correct type and format</p></td>
      </tr><tr><td>
          <p><code><a href="check_deg_method.html">check_deg_method()</a></code> </p>
        </td>
        <td><p>check if DEG method is one of the available options</p></td>
      </tr><tr><td>
          <p><code><a href="check_dim_selection_method.html">check_dim_selection_method()</a></code> </p>
        </td>
        <td><p>check dimension selection method</p></td>
      </tr><tr><td>
          <p><code><a href="check_embed_methods.html">check_embed_methods()</a></code> </p>
        </td>
        <td><p>check if embedding method is present in availble options</p></td>
      </tr><tr><td>
          <p><code><a href="check_embedding_selection.html">check_embedding_selection()</a></code> </p>
        </td>
        <td><p>check embedding selection</p></td>
      </tr><tr><td>
          <p><code><a href="check_embeddings.html">check_embeddings()</a></code> </p>
        </td>
        <td><p>check embeddings
checking user provided embedding matrix</p></td>
      </tr><tr><td>
          <p><code><a href="check_eq_method.html">check_eq_method()</a></code> </p>
        </td>
        <td><p>check if eq method are avilable</p></td>
      </tr><tr><td>
          <p><code><a href="check_features.html">check_features()</a></code> </p>
        </td>
        <td><p>check features</p></td>
      </tr><tr><td>
          <p><code><a href="check_for_zero_counts.html">check_for_zero_counts()</a></code> </p>
        </td>
        <td><p>check_for_zero_counts 
check which barcodes contain all 0 counts and filter</p></td>
      </tr><tr><td>
          <p><code><a href="check_group_value.html">check_group_value()</a></code> </p>
        </td>
        <td><p>check if requested territory is in territory list</p></td>
      </tr><tr><td>
          <p><code><a href="check_image.html">check_image()</a></code> </p>
        </td>
        <td><p>check if image is present</p></td>
      </tr><tr><td>
          <p><code><a href="check_image_input.html">check_image_input()</a></code> </p>
        </td>
        <td><p>check image input 
check if input image is in correct format</p></td>
      </tr><tr><td>
          <p><code><a href="check_inputs.html">check_inputs()</a></code> </p>
        </td>
        <td><p>check input checks the validity of input data to build_vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="check_isolation_method.html">check_isolation_method()</a></code> </p>
        </td>
        <td><p>check territory isolation method is available</p></td>
      </tr><tr><td>
          <p><code><a href="check_min_spatial_index.html">check_min_spatial_index()</a></code> </p>
        </td>
        <td><p>check number of spatial indices present</p></td>
      </tr><tr><td>
          <p><code><a href="check_norm.html">check_norm()</a></code> </p>
        </td>
        <td><p>check if select norm method is an available option</p></td>
      </tr><tr><td>
          <p><code><a href="check_norm_methods.html">check_norm_methods()</a></code> </p>
        </td>
        <td><p>check if norm method is present in availble options</p></td>
      </tr><tr><td>
          <p><code><a href="check_segment_trial.html">check_segment_trial()</a></code> </p>
        </td>
        <td><p>check if segment selection is a valid option</p></td>
      </tr><tr><td>
          <p><code><a href="check_segmentation_method.html">check_segmentation_method()</a></code> </p>
        </td>
        <td><p>check if segmentation options are valid for segmentation</p></td>
      </tr><tr><td>
          <p><code><a href="check_smoothing_kernel.html">check_smoothing_kernel()</a></code> </p>
        </td>
        <td><p>check if smoothing method is present in availble options</p></td>
      </tr><tr><td>
          <p><code><a href="check_smoothing_level.html">check_smoothing_level()</a></code> </p>
        </td>
        <td><p>check if across level options are valid for smoothing</p></td>
      </tr><tr><td>
          <p><code><a href="check_tensor_compression.html">check_tensor_compression()</a></code> </p>
        </td>
        <td><p>check tensor compression</p></td>
      </tr><tr><td>
          <p><code><a href="check_territory_trial.html">check_territory_trial()</a></code> </p>
        </td>
        <td><p>check if territory selection is a valid option</p></td>
      </tr><tr><td>
          <p><code><a href="check_tiles.html">check_tiles()</a></code> </p>
        </td>
        <td><p>check tile integretiy</p></td>
      </tr><tr><td>
          <p><code><a href="commit_log.html">commit_log()</a></code> </p>
        </td>
        <td><p>commits function call to log slot</p></td>
      </tr><tr><td>
          <p><code><a href="compute_effect_size.html">compute_effect_size()</a></code> </p>
        </td>
        <td><p>compute_effect_size</p></td>
      </tr><tr><td>
          <p><code><a href="compute_nearest_neighbor_graph.html">compute_nearest_neighbor_graph()</a></code> </p>
        </td>
        <td><p>compute and greate nearest neighbor graph</p></td>
      </tr><tr><td>
          <p><code><a href="compute_territory_area.html">compute_territory_area()</a></code> </p>
        </td>
        <td><p>Calculate the total area of a territory</p></td>
      </tr><tr><td>
          <p><code><a href="concat_embed.html">concat_embed()</a></code> </p>
        </td>
        <td><p>create new embedding from jointly measure spatial omics</p></td>
      </tr><tr><td>
          <p><code><a href="concat_matches.html">concat_matches()</a></code> </p>
        </td>
        <td><p>merging batch matches together</p></td>
      </tr><tr><td>
          <p><code><a href="convexify.html">convexify()</a></code> </p>
        </td>
        <td><p>convexify</p></td>
      </tr><tr><td>
          <p><code><a href="coordinates.html">vesalius</a></code> </p>
        </td>
        <td><p>Spatial coordinates</p></td>
      </tr><tr><td>
          <p><code><a href="cost_to_prob.html">cost_to_prob()</a></code> </p>
        </td>
        <td><p>convert cost to prob</p></td>
      </tr><tr><td>
          <p><code><a href="counts.html">vesalius</a></code> </p>
        </td>
        <td><p>Count matrix for vesalius</p></td>
      </tr><tr><td>
          <p><code><a href="create_alpha.html">create_alpha()</a></code> </p>
        </td>
        <td><p>create alpha value if territories need to be highlighted</p></td>
      </tr><tr><td>
          <p><code><a href="create_commit_log.html">create_commit_log()</a></code> </p>
        </td>
        <td><p>create function log to be commit to the log slot</p></td>
      </tr><tr><td>
          <p><code><a href="create_palette.html">create_palette()</a></code> </p>
        </td>
        <td><p>create color palette from predefine scheme</p></td>
      </tr><tr><td>
          <p><code><a href="create_pseudo_centroids.html">create_pseudo_centroids()</a></code> </p>
        </td>
        <td><p>create centroid vlues for louvain and leiden</p></td>
      </tr><tr><td>
          <p><code><a href="create_trial_tag.html">create_trial_tag()</a></code> </p>
        </td>
        <td><p>create a trail tag name</p></td>
      </tr><tr><td>
          <p><code><a href="detect_edges.html">detect_edges()</a></code> </p>
        </td>
        <td><p>detect_edges
detect territory edges in black and white images with sobel filter</p></td>
      </tr><tr><td>
          <p><code><a href="dispatch_deg_group.html">dispatch_deg_group()</a></code> </p>
        </td>
        <td><p>dispatch barcodes to seed and query groups</p></td>
      </tr><tr><td>
          <p><code><a href="dispatch_territory.html">dispatch_territory()</a></code> </p>
        </td>
        <td><p>dispatch territories labels territories according to which
group they belong to.</p></td>
      </tr><tr><td>
          <p><code><a href="distance_pooling.html">distance_pooling()</a></code> </p>
        </td>
        <td><p>distance pooling beads of colour segment into seperate territories</p></td>
      </tr><tr><td>
          <p><code><a href="divide_and_conquer.html">divide_and_conquer()</a></code> </p>
        </td>
        <td><p>splitting cost matrix into batches for batch processing</p></td>
      </tr><tr><td>
          <p><code><a href="ecdf_eq.html">ecdf_eq()</a></code> </p>
        </td>
        <td><p>internal ecdf eq 
used for formatting</p></td>
      </tr><tr><td>
          <p><code><a href="embed_latent_space.html">embed_latent_space()</a></code> </p>
        </td>
        <td><p>embed latent space</p></td>
      </tr><tr><td>
          <p><code><a href="embed_lsi.html">embed_lsi()</a></code> </p>
        </td>
        <td><p>embed lsi</p></td>
      </tr><tr><td>
          <p><code><a href="embed_lsi_umap.html">embed_lsi_umap()</a></code> </p>
        </td>
        <td><p>embed lsi</p></td>
      </tr><tr><td>
          <p><code><a href="embed_nmf.html">embed_nmf()</a></code> </p>
        </td>
        <td><p>embed nmf</p></td>
      </tr><tr><td>
          <p><code><a href="embed_pca.html">embed_pca()</a></code> </p>
        </td>
        <td><p>embed PCA</p></td>
      </tr><tr><td>
          <p><code><a href="embed_pcal.html">embed_pcal()</a></code> </p>
        </td>
        <td><p>embed PCA loading values</p></td>
      </tr><tr><td>
          <p><code><a href="embed_umap.html">embed_umap()</a></code> </p>
        </td>
        <td><p>embed umap</p></td>
      </tr><tr><td>
          <p><code><a href="equalize_image.html">equalize_image()</a></code> </p>
        </td>
        <td><p>equalise image histogram</p></td>
      </tr><tr><td>
          <p><code><a href="extend_boundary.html">extend_boundary()</a></code> </p>
        </td>
        <td><p>extend image boundary around territory</p></td>
      </tr><tr><td>
          <p><code><a href="feature_dist.html">feature_dist()</a></code> </p>
        </td>
        <td><p>compute the distance between seed and query signals</p></td>
      </tr><tr><td>
          <p><code><a href="filter_grid.html">filter_grid()</a></code> </p>
        </td>
        <td><p>filter grid</p></td>
      </tr><tr><td>
          <p><code><a href="filter_tiles.html">filter_tiles()</a></code> </p>
        </td>
        <td><p>filter tiles</p></td>
      </tr><tr><td>
          <p><code><a href="format_c_to_ves.html">format_c_to_ves()</a></code> </p>
        </td>
        <td><p>convert cimg list to vesalius object embedding</p></td>
      </tr><tr><td>
          <p><code><a href="format_call.html">format_call()</a></code> </p>
        </td>
        <td><p>format function call to list for log update</p></td>
      </tr><tr><td>
          <p><code><a href="format_counts_for_deseq2.html">format_counts_for_deseq2()</a></code> </p>
        </td>
        <td><p>format counts for DESeq</p></td>
      </tr><tr><td>
          <p><code><a href="format_counts_for_edger.html">format_counts_for_edger()</a></code> </p>
        </td>
        <td><p>format counts for edgeR</p></td>
      </tr><tr><td>
          <p><code><a href="format_counts_for_logit.html">format_counts_for_logit()</a></code> </p>
        </td>
        <td><p>format counts for logistic regression</p></td>
      </tr><tr><td>
          <p><code><a href="format_ves_to_c.html">format_ves_to_c()</a></code> </p>
        </td>
        <td><p>convert vesalius_assay to cimg images</p></td>
      </tr><tr><td>
          <p><code><a href="future_ves_to_cimg.html">future_ves_to_cimg()</a></code> </p>
        </td>
        <td><p>convert ves embedding to image</p></td>
      </tr><tr><td>
          <p><code><a href="generate_embeddings.html">generate_embeddings()</a></code> </p>
        </td>
        <td><p>Generate embeddings.</p></td>
      </tr><tr><td>
          <p><code><a href="generate_tiles.html">generate_tiles()</a></code> </p>
        </td>
        <td><p>generate tiles</p></td>
      </tr><tr><td>
          <p><code><a href="get_active_count_tag.html">get_active_count_tag()</a></code> </p>
        </td>
        <td><p>get last count matrix used</p></td>
      </tr><tr><td>
          <p><code><a href="get_active_embedding_tag.html">get_active_embedding_tag()</a></code> </p>
        </td>
        <td><p>get active embedding</p></td>
      </tr><tr><td>
          <p><code><a href="get_assay_names.html">get_assay_names()</a></code> </p>
        </td>
        <td><p>get assay names from vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="get_coordinayes.html">get_coordinates()</a></code> </p>
        </td>
        <td><p>get coordinates from vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="get_counts.html">get_counts()</a></code> </p>
        </td>
        <td><p>get counts from vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="get_deg_metrics.html">get_deg_metrics()</a></code> </p>
        </td>
        <td><p>get DEG metrics from groups</p></td>
      </tr><tr><td>
          <p><code><a href="get_embeddings.html">get_embeddings()</a></code> </p>
        </td>
        <td><p>get embeddings from vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="get_features.html">get_features()</a></code> </p>
        </td>
        <td><p>get cell signal from vesalius assays</p></td>
      </tr><tr><td>
          <p><code><a href="get_func_from_commit.html">get_func_from_commit()</a></code> </p>
        </td>
        <td><p>get function name from commit list</p></td>
      </tr><tr><td>
          <p><code><a href="get_markers.html">get_markers()</a></code> </p>
        </td>
        <td><p>get markers from vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="get_territories.html">get_territories()</a></code> </p>
        </td>
        <td><p>get territories from vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="get_tiles.html">get_tiles()</a></code> </p>
        </td>
        <td><p>get tiles from vesalius_assay</p></td>
      </tr><tr><td>
          <p><code><a href="globalise_territories.html">globalise_territories()</a></code> </p>
        </td>
        <td><p>globalise territories
converts territories number from territory values per segment
to territory values overall the entire spatial omics assay.</p></td>
      </tr><tr><td>
          <p><code><a href="identify_markers.html">identify_markers()</a></code> </p>
        </td>
        <td><p>identify_markers computes differential observation expression 
between selected territories.</p></td>
      </tr><tr><td>
          <p><code><a href="image_plot.html">image_plot()</a></code> </p>
        </td>
        <td><p>image_plot - plotting vesalius embeddings</p></td>
      </tr><tr><td>
          <p><code><a href="int_sctransform.html">int_sctransform()</a></code> </p>
        </td>
        <td><p>SCTransform</p></td>
      </tr><tr><td>
          <p><code><a href="interlace_embeds.html">interlace_embeds()</a></code> </p>
        </td>
        <td><p>interlace image stack between seed and query</p></td>
      </tr><tr><td>
          <p><code><a href="internal_smooth.html">internal_smooth()</a></code> </p>
        </td>
        <td><p>internal smoothing function
Function that does the smoothing on individual image array</p></td>
      </tr><tr><td>
          <p><code><a href="isolate_territories.html">isolate_territories()</a></code> </p>
        </td>
        <td><p>isolating territories from vesalius image segments</p></td>
      </tr><tr><td>
          <p><code><a href="joint_territories.html">joint_territories()</a></code> </p>
        </td>
        <td><p>Jointly measured spatial omic assays territories</p></td>
      </tr><tr><td>
          <p><code><a href="kmeans_segmentation.html">kmeans_segmentation()</a></code> </p>
        </td>
        <td><p>kmeans segmentation function</p></td>
      </tr><tr><td>
          <p><code><a href="layer_territory.html">layer_territory()</a></code> </p>
        </td>
        <td><p>layer_territory generates layer from the outside to the inside of 
a territory</p></td>
      </tr><tr><td>
          <p><code><a href="leiden_segmentation.html">leiden_segmentation()</a></code> </p>
        </td>
        <td><p>leiden segmentation</p></td>
      </tr><tr><td>
          <p><code><a href="log_norm.html">log_norm()</a></code> </p>
        </td>
        <td><p>log norm</p></td>
      </tr><tr><td>
          <p><code><a href="louvain_segmentation.html">louvain_segmentation()</a></code> </p>
        </td>
        <td><p>louvain segmentation</p></td>
      </tr><tr><td>
          <p><code><a href="map_assays.html">map_assays()</a></code> </p>
        </td>
        <td><p>Aling and integrate spatial assay from the same modality using super pixels</p></td>
      </tr><tr><td>
          <p><code><a href="match_index.html">match_index()</a></code> </p>
        </td>
        <td><p>Finding optimal mapping between query and seed using a mini batch 
approach</p></td>
      </tr><tr><td>
          <p><code><a href="message_switch.html">message_switch()</a></code> </p>
        </td>
        <td><p>switch message output based in input</p></td>
      </tr><tr><td>
          <p><code><a href="min_max.html">min_max()</a></code> </p>
        </td>
        <td><p>min max normalisation</p></td>
      </tr><tr><td>
          <p><code><a href="neighborhood_signal.html">neighborhood_signal()</a></code> </p>
        </td>
        <td><p>compute average expression of local neighborhood</p></td>
      </tr><tr><td>
          <p><code><a href="norm_pixel.html">norm_pixel()</a></code> </p>
        </td>
        <td><p>pixel normalisation dispatch function</p></td>
      </tr><tr><td>
          <p><code><a href="point_mapping.html">point_mapping()</a></code> </p>
        </td>
        <td><p>mapping points between data sets</p></td>
      </tr><tr><td>
          <p><code><a href="populate_graph.html">populate_graph()</a></code> </p>
        </td>
        <td><p>populate graph with network connections</p></td>
      </tr><tr><td>
          <p><code><a href="process_counts.html">process_counts()</a></code> </p>
        </td>
        <td><p>process counts</p></td>
      </tr><tr><td>
          <p><code><a href="rasterise.html">rasterise()</a></code> </p>
        </td>
        <td><p>rasterise tiles</p></td>
      </tr><tr><td>
          <p><code><a href="raw_norm.html">raw_norm()</a></code> </p>
        </td>
        <td><p>raw norm</p></td>
      </tr><tr><td>
          <p><code><a href="rebalence_colors.html">rebalence_colors()</a></code> </p>
        </td>
        <td><p>rebalence colors</p></td>
      </tr><tr><td>
          <p><code><a href="reduce_tensor_resolution.html">reduce_tensor_resolution()</a></code> </p>
        </td>
        <td><p>reduce tensor resoltuon</p></td>
      </tr><tr><td>
          <p><code><a href="regularise.html">regularise()</a></code> </p>
        </td>
        <td><p>Internal regularise function.
performs total variance regularisation</p></td>
      </tr><tr><td>
          <p><code><a href="regularise_image.html">regularise_image()</a></code> </p>
        </td>
        <td><p>regularise image</p></td>
      </tr><tr><td>
          <p><code><a href="search_log.html">search_log()</a></code> </p>
        </td>
        <td><p>search through log for parameter values or names</p></td>
      </tr><tr><td>
          <p><code><a href="segment_image.html">segment_image()</a></code> </p>
        </td>
        <td><p>segment image</p></td>
      </tr><tr><td>
          <p><code><a href="select_initial_indices.html">select_initial_indices()</a></code> </p>
        </td>
        <td><p>select inital indices</p></td>
      </tr><tr><td>
          <p><code><a href="smooth_image.html">smooth_image()</a></code> </p>
        </td>
        <td><p>Smooth Image</p></td>
      </tr><tr><td>
          <p><code><a href="summarise_territories.html">summarise_territories()</a></code> </p>
        </td>
        <td><p>summarise territories</p></td>
      </tr><tr><td>
          <p><code><a href="territory_morphing.html">territory_morphing()</a></code> </p>
        </td>
        <td><p>territory_morphing applies morphological operators to a set of territoriees</p></td>
      </tr><tr><td>
          <p><code><a href="territory_plot.html">territory_plot()</a></code> </p>
        </td>
        <td><p>territory_plot - plotting Vesalius territories</p></td>
      </tr><tr><td>
          <p><code><a href="tfidf_norm.html">tfidf_norm()</a></code> </p>
        </td>
        <td><p>tf idf normalisation</p></td>
      </tr><tr><td>
          <p><code><a href="unpack_territory_path.html">unpack_territory_path()</a></code> </p>
        </td>
        <td><p>retrieve the points contained in the edge of each territory</p></td>
      </tr><tr><td>
          <p><code><a href="update_vesalius_assay.html">update_vesalius_assay()</a></code> </p>
        </td>
        <td><p>update vesalius assay object</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_assay-class.html">vesalius_assay-class</a></code> </p>
        </td>
        <td><p>The vesalius_assay class</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg.html">vesalius_deg()</a></code> </p>
        </td>
        <td><p>Internal differantial gene expression function.</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg_chisq.html">vesalius_deg_chisq()</a></code> </p>
        </td>
        <td><p>chisq test for DEG</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg_deseq2.html">vesalius_deg_deseq2()</a></code> </p>
        </td>
        <td><p>DESeq negatove binomail for DEG</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg_edger.html">vesalius_deg_edger()</a></code> </p>
        </td>
        <td><p>edgeR functions for DEG</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg_fisher.html">vesalius_deg_fisher()</a></code> </p>
        </td>
        <td><p>Fisher's excat test for DEG</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg_logit.html">vesalius_deg_logit()</a></code> </p>
        </td>
        <td><p>Logistic Regression for DEG</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg_ttest.html">vesalius_deg_ttest()</a></code> </p>
        </td>
        <td><p>t.test for DEG</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_deg_wilcox.html">vesalius_deg_wilcox()</a></code> </p>
        </td>
        <td><p>wilcox rank sum test for DEG</p></td>
      </tr><tr><td>
          <p><code><a href="vesalius_package.html">vesalius_package</a></code> <code><a href="vesalius_package.html">vesalius-package</a></code> </p>
        </td>
        <td><p>vesalius</p></td>
      </tr><tr><td>
          <p><code><a href="view_gene_expression.html">view_gene_expression()</a></code> </p>
        </td>
        <td><p>view_gene_expression</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Patrick C.N. Martin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

