<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Vesalius: Quick Start • vesalius</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Vesalius: Quick Start">
<meta property="og:description" content="vesalius">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">vesalius</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/vesalius.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vesalius_Analysis/Spatial_DNA_seq.html">Vesalius: Spatial Genomics</a>
    </li>
    <li>
      <a href="../articles/Vesalius_Analysis/Spatial_Transcriptomics.html">Vesalius: Spatial Transcriptomics</a>
    </li>
    <li>
      <a href="../articles/Vesalius_Analysis/Vesalius_MSB_analysis.html">Vesalius: MSB Manuscript Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/patrickCNMartin/Vesalius/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Vesalius: Quick Start</h1>
                        <h4 data-toc-skip class="author">Patrick C.N.
Martin</h4>
            
            <h4 data-toc-skip class="date">2024-01-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/patrickCNMartin/Vesalius/blob/main/vignettes/vesalius.Rmd" class="external-link"><code>vignettes/vesalius.Rmd</code></a></small>
      <div class="hidden name"><code>vesalius.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="vesalius-quick-start">Vesalius: Quick Start<a class="anchor" aria-label="anchor" href="#vesalius-quick-start"></a>
</h2>
<p>Do you like to live in the fast lane? Do you sigh every time you see
a long vignette? Do you wish there was a TLDR on everything?</p>
<p>I understand.</p>
<p>Welcome to the Vesalius quick start!</p>
<p>In this vignette, you will find a <strong>very</strong> short guide
on how to work with Vesalius. Vesalius provides an internal data set
taken from real Spatial transcrtiptomic data.</p>
<p>We show <em>in context</em> Vesalius workflows in the articles
section.</p>
<p>For all intents and purposes, this is a dummy data set and should
only be used to get a feel for the Vesalius package. We strongly advise
you to use real and complete data sets to fully understand the Vesalius
package and its benefits.</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Vesalius is a tool to perform high-resolution in silico anatomization
and molecular characterization from Spatial omices data without
requiring companion images. The nature of the vesalius algorithm makes
it condusive to being using on various spatial omics modalities without
requiuring substantial changes.</p>
<p>Vesalius achieves this by converting reducded dimensionality latent
space into gray scale images (one of each dimension). These images are
processed using various image analysis techniques such as smoothing and
segmentation.</p>
</div>
<div class="section level2">
<h2 id="loading-building-a-vesalius-assay">Loading &amp; building a vesalius assay<a class="anchor" aria-label="anchor" href="#loading-building-a-vesalius-assay"></a>
</h2>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>First, let’s load the package and the data. Data originates from
slide-seqV2 data (availble at <a href="https://singlecell.broadinstitute.org/single_cell/study/SCP948/robust-decomposition-of-cell-type-mixtures-in-spatial-transcriptomics#study-download" class="external-link">Single
Cell Portal</a>).</p>
<p>We took a small subset of the mouse hippocampus centered around the
Dentate Gyrus.</p>
<p>Here, we show an example using transcriptomic data but the same
principles will apply for other modalities. For examples using other
modalities, please refer to the other articles.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patrickcnmartin.github.io/Vesalius/">vesalius</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">vesalius</span>, package <span class="op">=</span> <span class="st">"vesalius"</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="what-do-we-have">What do we have?<a class="anchor" aria-label="anchor" href="#what-do-we-have"></a>
</h4>
<p>Once loaded we have two new variables: counts and cooridnates.</p>
<p>The counts are the gene counts for all genes. As per convention,
columns r epresent the spatial indices (barcodes, spots, beads,…) and
rows are the genes that were captured.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span>
<span><span class="co">##   ..@ i       : int [1:653622] 4 8 10 15 18 19 30 38 41 51 ...</span></span>
<span><span class="co">##   ..@ p       : int [1:1053] 0 2982 6344 9623 12651 15540 18593 21462 24241 26859 ...</span></span>
<span><span class="co">##   ..@ Dim     : int [1:2] 15932 1052</span></span>
<span><span class="co">##   ..@ Dimnames:List of 2</span></span>
<span><span class="co">##   .. ..$ : chr [1:15932] "0610005C13Rik" "0610007P14Rik" "0610009B22Rik" "0610009E02Rik" ...</span></span>
<span><span class="co">##   .. ..$ : chr [1:1052] "CTCCAGCCTTGCAT" "AACCCCAAGGTAAT" "GCCCAACACACTTT" "GTCCAGAGTCTTAT" ...</span></span>
<span><span class="co">##   ..@ x       : num [1:653622] 1 4 1 2 2 2 2 1 1 1 ...</span></span>
<span><span class="co">##   ..@ factors : list()</span></span></code></pre>
<p>The coordinates represent the x and y coordinates for each spatial
index.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">coordinates</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'data.frame':    1052 obs. of  3 variables:</span></span>
<span><span class="co">##  $ barcodes: chr  "CTCCAGCCTTGCAT" "AACCCCAAGGTAAT" "GCCCAACACACTTT" "GTCCAGAGTCTTAT" ...</span></span>
<span><span class="co">##  $ xcoord  : num  3536 3472 3634 3589 3535 ...</span></span>
<span><span class="co">##  $ ycoord  : num  3485 3410 3636 3257 3352 ...</span></span></code></pre>
<p>In other modalities, count matrices are still required for vesalius
to run.</p>
<p>For example, Spatial-ATAC-seq or Spatial-cut&amp;tag can be converted
into count matrices (more specifically tile matrices) using the <a href="https://www.archrproject.com/" class="external-link">ArchR package</a></p>
</div>
</div>
<div class="section level3">
<h3 id="building-a-vesalius-assay">Building a Vesalius assay<a class="anchor" aria-label="anchor" href="#building-a-vesalius-assay"></a>
</h3>
<p>We can build a vesalius assay by simply parsing both counts and
cooridnates to the (vesalius assay contructor)[link to manual page].</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_vesalius_assay.html">build_vesalius_assay</a></span><span class="op">(</span></span>
<span>  coordinates <span class="op">=</span> <span class="va">coordinates</span>, <span class="co"># spatial coordinates</span></span>
<span>  counts  <span class="op">=</span> <span class="va">counts</span>, <span class="co"># count matrix</span></span>
<span>  assay <span class="op">=</span> <span class="st">"spatial_omics"</span>, <span class="co"># name you wish to give your assay</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Do you want progress messages?</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="what-do-we-have-1">What do we have?<a class="anchor" aria-label="anchor" href="#what-do-we-have-1"></a>
</h4>
<p>A vesalius object is a container that requires at the very least
spatial indices. Here, we parsed both counts and spatial indices</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## spatial_omics as vesalius assay containing:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1 modifications applied to base object. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1052 spatial indices. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 15932 observations in the raw count matrix. </span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<p>Count matrices can be added sperately using the
<code>add_counts</code> function. For more information, please refere to
the <a href="">Modifying vesalius objects</a></p>
</div>
</div>
<div class="section level3">
<h3 id="interoperability">Interoperability<a class="anchor" aria-label="anchor" href="#interoperability"></a>
</h3>
<p>UNDER CONSTRCUTION!</p>
<p>We are currently working on making Vesalis fully compatible with
SpatialFeatureExperiments, SingleCellExperiment and anndata.</p>
</div>
</div>
<div class="section level2">
<h2 id="embedding-latent-space-into-gray-scale-images">Embedding latent space into gray scale images<a class="anchor" aria-label="anchor" href="#embedding-latent-space-into-gray-scale-images"></a>
</h2>
<div class="section level3">
<h3 id="embed-latent-space">Embed Latent Space<a class="anchor" aria-label="anchor" href="#embed-latent-space"></a>
</h3>
<p>We can use this object top embed our latent space into grey scale
images.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_embeddings.html">generate_embeddings</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  dim_reduction <span class="op">=</span> <span class="st">"PCA"</span>,</span>
<span>  normalisation <span class="op">=</span> <span class="st">"log_norm"</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">100</span>, <span class="co"># Setting number of features low for low run time</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="what-do-we-have-2">What do we have?<a class="anchor" aria-label="anchor" href="#what-do-we-have-2"></a>
</h4>
<p>We see that we have added a set of embedding with one used one
default embedding called the “active” embedding. See section below on
<em>understanding active embeddings</em>.</p>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## spatial_omics as vesalius assay containing:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 3 modifications applied to base object. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1052 spatial indices used to form pixel tiles. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 15932 observations in the log_norm count matrix. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## PCA as embeddings. </span></span>
<span><span class="co">## with PCA as active embedding. </span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="testing-multiple-embeddings">Testing multiple embeddings<a class="anchor" aria-label="anchor" href="#testing-multiple-embeddings"></a>
</h3>
<p>We can test out multiple embeddings and every trial will be stored
into the <code>vesalius_assay</code> object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_embeddings.html">generate_embeddings</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  dim_reduction <span class="op">=</span> <span class="st">"UMAP"</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">100</span>, <span class="co"># Setting number of features low for low run time</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'spam'</span></span></code></pre>
<pre><code><span><span class="co">## Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'spam'</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_embeddings.html">generate_embeddings</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  dim_reduction <span class="op">=</span> <span class="st">"PCA"</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">200</span>, <span class="co"># Setting number of features low for low run time</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_embeddings.html">generate_embeddings</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  dim_reduction <span class="op">=</span> <span class="st">"PCA"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Every time you run a new dimensionality reduction approach it will be
appended to the previous ones. Vesalius does the same thing for
different normalisation approaches. This lets you decided which
combination you would want to use in later stages.</p>
<p>Note that since you might be want to use the same embedding more than
once, you will end up with more than one embedding called “PCA” for
example. In this case, each embedding or normalisation will have a
unique name.</p>
<p>For every subsequent embedding called PCA, the trial name will be
PCA.1, PCA.2 etc The same applies to any embedding or normalisation
method you select.</p>
<div class="section level4">
<h4 id="what-do-we-have-3">What do we have ?<a class="anchor" aria-label="anchor" href="#what-do-we-have-3"></a>
</h4>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## spatial_omics as vesalius assay containing:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 6 modifications applied to base object. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1052 spatial indices used to form pixel tiles. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 15932 observations in the log_norm.3 count matrix. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 4 total embedding trials</span></span>
<span><span class="co">## with PCA.2 as active embedding. </span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<p>You can also add your own embeddings and they will be appended into
vesalius in a similar manner. For more information, please refere to the
<a href="">modifying vesalius objects</a> vignette. Alternively, you can
also check the manual page for <a href="">add_embeddings</a></p>
</div>
</div>
<div class="section level3">
<h3 id="visualising-grey-scale-images">Visualising Grey scale images<a class="anchor" aria-label="anchor" href="#visualising-grey-scale-images"></a>
</h3>
<p>To visualise embeddings, we can simply plot the vesalius object and
specify which dimension we wish to view. See (imagePlot)[link to page]
for more information. Since we are leaving the embedding argument as the
default “last” value, we can only look at the 3 UMAP dimensions. We can
look at them simulaneously by creating RGB images with one dimension per
color channel.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/image_plot.html">image_plot</a></span><span class="op">(</span><span class="va">vesalius</span>, dimensions <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Grey PCA dim 1"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/image_plot.html">image_plot</a></span><span class="op">(</span><span class="va">vesalius</span>, dimensions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"RGB PCA"</span><span class="op">)</span></span></code></pre></div>
<p>If we wanted to look at “UMAP” instead, we can simply specify the
embedding. Since we used 30 dimensions for UMAP, we can use any
dimensions to build RGB images.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/image_plot.html">image_plot</a></span><span class="op">(</span><span class="va">vesalius</span>, dimensions <span class="op">=</span> <span class="fl">1</span>, embedding <span class="op">=</span> <span class="st">"UMAP"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Grey UMAP dim 1"</span><span class="op">)</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/image_plot.html">image_plot</a></span><span class="op">(</span><span class="va">vesalius</span>, dimensions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, embedding <span class="op">=</span> <span class="st">"UMAP"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Grey UMAP dim 1, 2, and 3"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="va">p3</span> <span class="op">+</span> <span class="va">p4</span><span class="op">)</span></span></code></pre></div>
<p><img src="vesalius_files/figure-html/out_plot_grey-1.png" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="understanding-active-embeddings">Understanding Active Embeddings<a class="anchor" aria-label="anchor" href="#understanding-active-embeddings"></a>
</h2>
<p>The active embedding slot in a <code>vesalius_assay</code> object
contains the embedding values that will be used during image processing
and image segmentation.</p>
<p>In the examples above, we tested multiple embeddings.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_active_embedding_tag.html">get_active_embedding_tag</a></span><span class="op">(</span><span class="va">vesalius</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "PCA.2"</span></span></code></pre>
<p>We see that “UMAP.1” is used as the active embedding. Any time the
embeddings are being used in subsequent functions they will be taken
from here as this is the “last” embedding used.</p>
<p>In fact, in any subsequent function that contains the keyword “last”
for one of its arguments, vesalius will use the last target created.
This can mean the last embedding used, the last count matrix or the last
segmentation / territory isolation / territory morphology used.</p>
<p>If you are unstaified with the results, you can always call the
embedding or trial you want to use explictely and the active slot will
be replace with a fresh version of your embeddings of choice. No need to
re-run everything from scratch!</p>
<p>While this flexibility can be a boon, it can also be a curse! Anytime
you explictely request an embedding or a trial it will use that fresh
instance. This means that every time the active slot will be updated and
any processing you may have done to it will be overwritten.</p>
</div>
<div class="section level2">
<h2 id="image-processing-and-image-segmentation">Image processing and Image segmentation<a class="anchor" aria-label="anchor" href="#image-processing-and-image-segmentation"></a>
</h2>
<div class="section level3">
<h3 id="image-processing">Image Processing<a class="anchor" aria-label="anchor" href="#image-processing"></a>
</h3>
<p>Once we have these images, we can apply image processing techniques
to each gray scale image.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/regularise_image.html">regularise_image</a></span><span class="op">(</span><span class="va">vesalius</span>, lambda <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## 2024-01-12 18:32:53.614659  Converting Vesalius to Image</span></span>
<span><span class="co">## 2024-01-12 18:32:54.261911  Regularising Image </span></span>
<span><span class="co">## 2024-01-12 18:32:58.042307  Converting Images to Vesalius</span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/smooth_image.html">smooth_image</a></span><span class="op">(</span><span class="va">vesalius</span>, sigma <span class="op">=</span> <span class="fl">5</span>, iter <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## 2024-01-12 18:32:58.296001  Converting Vesalius to Image</span></span>
<span><span class="co">## 2024-01-12 18:32:59.635372  Smoothing Image Arrays </span></span>
<span><span class="co">## 2024-01-12 18:32:59.890314  Converting Images to Vesalius</span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/equalize_image.html">equalize_image</a></span><span class="op">(</span><span class="va">vesalius</span>, sleft <span class="op">=</span> <span class="fl">5</span>, sright <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## 2024-01-12 18:33:00.085996  Converting Vesalius to Image</span></span>
<span><span class="co">## 2024-01-12 18:33:01.336601  Equalizing Histogram </span></span>
<span><span class="co">## 2024-01-12 18:33:01.414701  Converting Images to Vesalius</span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<p>You can apply any image processing method you wish and in any order
you desire. Please note that the default number of dimensions for these
function is defined by <code>seq(1,3)</code> giving the first three
dimensions. If working with PCA, please set this argument according to
how many dimensions you wish to process.</p>
<p>This also alows for the selction of abitrary of PCs. For example,
based on the grey scale images, you might decided that this would be a
better choice of PCs</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Selecting a subset of PCs</span></span>
<span><span class="va">dims</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">7</span><span class="op">:</span><span class="fl">11</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># running smoothing o</span></span>
<span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/regularise_image.html">regularise_image</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  dimensions <span class="op">=</span> <span class="va">dims</span>,</span>
<span>  embedding <span class="op">=</span> <span class="st">"PCA"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/equalize_image.html">equalize_image</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  dimensions <span class="op">=</span> <span class="va">dims</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/smooth_image.html">smooth_image</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  dimensions <span class="op">=</span> <span class="va">dims</span>,</span>
<span>  iter <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  sigma <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>In this example, we explicetely requested the <em>PCA</em> embedding.
This means that we will take a fresh instance of the <em>PCA</em>
embedding and use this as our active embedding. Once specified, all
subsequent functions will use this new active embedding unless specified
otherwise.</p>
<p>However, note that dimensions are being parsed at every step. Only
the images coressponding to these dimensions will be processed. Vesalius
still retains the other ones just in case.</p>
</div>
<div class="section level3">
<h3 id="image-segmentation">Image segmentation<a class="anchor" aria-label="anchor" href="#image-segmentation"></a>
</h3>
<p>Vesalius will attempts to segment the images into color segments.
Vesalius use a kmeans clustering approach to segment grey scale images.
Note that the segmentation here is applied to the whole stack and not
individually.</p>
<p>The goal is top obtain colour segments that we can subdivived into
territories.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/segment_image.html">segment_image</a></span><span class="op">(</span><span class="va">vesalius</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"kmeans"</span>,</span>
<span>  col_resolution <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="what-do-we-have-4">What do we have?<a class="anchor" aria-label="anchor" href="#what-do-we-have-4"></a>
</h3>
<p>We can see that the vesalius_assay object now contains some more
information related to the segmentation.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## spatial_omics as vesalius assay containing:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 10 modifications applied to base object. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1052 spatial indices used to form pixel tiles. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 15932 observations in the log_norm.3 count matrix. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 4 total embedding trials</span></span>
<span><span class="co">## with PCA.2 as active embedding. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1 Segmentation trials. </span></span>
<span><span class="co">## 0 Territory trials. </span></span>
<span><span class="co">## 0 Morphing trials. </span></span>
<span><span class="co">## 0 Layering trials. </span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<p>We can also have a look at the results of the image segmentation.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p5</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/image_plot.html">image_plot</a></span><span class="op">(</span><span class="va">vesalius</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Segments only"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p5</span><span class="op">)</span></span></code></pre></div>
<p><img src="vesalius_files/figure-html/plot_segments_only-1.png" width="960"></p>
<p>Kmeans is the default setting for image segmentation but Vesalius
also provides <em>louvain</em> and <em>leiden</em> based approaches.</p>
</div>
</div>
<div class="section level2">
<h2 id="isolating-territories">Isolating territories<a class="anchor" aria-label="anchor" href="#isolating-territories"></a>
</h2>
<div class="section level3">
<h3 id="isolating-territories-from-colour-segments">Isolating territories from colour segments<a class="anchor" aria-label="anchor" href="#isolating-territories-from-colour-segments"></a>
</h3>
<p>The final step is to isolate color segments into seperate
territories. Similar color segments may be of the same colour but are
seperated in 2D space. We want to be able to isolate each patch. For
this, we can use the <code>isolate_territories</code> function.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/isolate_territories.html">isolate_territories</a></span><span class="op">(</span><span class="va">vesalius</span>, capture_radius <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## 2024-01-12 18:33:04.068757  Pooling Segment  1 </span></span>
<span><span class="co">## 2024-01-12 18:33:04.077503  Pooling Segment  2 </span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<div class="section level4">
<h4 id="what-do-we-have-5">What do we have?<a class="anchor" aria-label="anchor" href="#what-do-we-have-5"></a>
</h4>
<p>Now we can have a look at the isolated territories. We can keep track
of how many instances of image segmentation and image manipulation we
have gone through.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## spatial_omics as vesalius assay containing:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 11 modifications applied to base object. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1052 spatial indices used to form pixel tiles. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 15932 observations in the log_norm.3 count matrix. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 4 total embedding trials</span></span>
<span><span class="co">## with PCA.2 as active embedding. </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 1 Segmentation trials. </span></span>
<span><span class="co">## 1 Territory trials. </span></span>
<span><span class="co">## 0 Morphing trials. </span></span>
<span><span class="co">## 0 Layering trials. </span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="visualizing-territories">Visualizing Territories<a class="anchor" aria-label="anchor" href="#visualizing-territories"></a>
</h3>
<p>We can plot our territories using vesalius plotting functions. Note
that this is a <code>ggplot</code> object as such it is customisable
using <code>ggplot2</code> functionalities.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p6</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/territory_plot.html">territory_plot</a></span><span class="op">(</span><span class="va">vesalius</span>, cex_pt <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span>
<span><span class="va">p6</span></span></code></pre></div>
<p><img src="vesalius_files/figure-html/plot_territories-1.png" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="comparing-territories">Comparing territories<a class="anchor" aria-label="anchor" href="#comparing-territories"></a>
</h2>
<p>Once we have our territories, we can compare the expression of genes
between territories.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vesalius</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/identify_markers.html">identify_markers</a></span><span class="op">(</span><span class="va">vesalius</span>, seed <span class="op">=</span> <span class="fl">1</span>, query <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## #--------------------------------------------------------------------------------# </span></span>
<span><span class="co">## 2024-01-12 18:33:04.958376 ===&gt; Computing DEGs in spatial_omics &lt;===</span></span>
<span><span class="co">## 2024-01-12 18:33:05.706732 ===&gt; 1 VS 2 &lt;=== </span></span>
<span><span class="co">## #--------------------------------------------------------------------------------#</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_markers.html">get_markers</a></span><span class="op">(</span><span class="va">vesalius</span><span class="op">)</span></span></code></pre></div>
<p>And finally we can visualise the expression of our genes of interest.
You can parse more than one gene. Here we show the overall expression
profile of <strong>Malat1</strong> but also its mean expression in each
territory.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/view_gene_expression.html">view_gene_expression</a></span><span class="op">(</span><span class="va">vesalius</span>, genes <span class="op">=</span> <span class="st">"Malat1"</span><span class="op">)</span></span>
<span><span class="va">p8</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/view_gene_expression.html">view_gene_expression</a></span><span class="op">(</span><span class="va">vesalius</span>, genes <span class="op">=</span> <span class="st">"Malat1"</span>, as_layer <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p7</span> <span class="op">+</span> <span class="va">p8</span></span></code></pre></div>
<p><img src="vesalius_files/figure-html/view_DEG-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="see-also">See Also<a class="anchor" aria-label="anchor" href="#see-also"></a>
</h2>
<p>For more information about different aspects of Vesalius:</p>
<ul>
<li><a href="">Modifying objects</a></li>
<li><a href="">Understanding vesalius_assay object</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ggplot2_3.4.4       patchwork_1.2.0     vesalius_2.0.0     </span></span>
<span><span class="co">## [4] Biobase_2.62.0      BiocGenerics_0.48.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RcppAnnoy_0.0.21            splines_4.3.2              </span></span>
<span><span class="co">##   [3] later_1.3.2                 bitops_1.0-7               </span></span>
<span><span class="co">##   [5] tibble_3.2.1                polyclip_1.10-6            </span></span>
<span><span class="co">##   [7] fastDummies_1.7.3           lifecycle_1.0.4            </span></span>
<span><span class="co">##   [9] Rdpack_2.6                  rstatix_0.7.2              </span></span>
<span><span class="co">##  [11] edgeR_4.0.6                 doParallel_1.0.17          </span></span>
<span><span class="co">##  [13] globals_0.16.2              lattice_0.21-9             </span></span>
<span><span class="co">##  [15] MASS_7.3-60                 readbitmap_0.1.5           </span></span>
<span><span class="co">##  [17] backports_1.4.1             magrittr_2.0.3             </span></span>
<span><span class="co">##  [19] limma_3.58.1                plotly_4.10.3              </span></span>
<span><span class="co">##  [21] sass_0.4.8                  rmarkdown_2.25             </span></span>
<span><span class="co">##  [23] jquerylib_0.1.4             yaml_2.3.8                 </span></span>
<span><span class="co">##  [25] httpuv_1.6.13               Seurat_5.0.1               </span></span>
<span><span class="co">##  [27] NMF_0.26                    sctransform_0.4.1          </span></span>
<span><span class="co">##  [29] spam_2.10-0                 sp_2.1-2                   </span></span>
<span><span class="co">##  [31] spatstat.sparse_3.0-3       reticulate_1.34.0          </span></span>
<span><span class="co">##  [33] cowplot_1.1.2               pbapply_1.7-2              </span></span>
<span><span class="co">##  [35] RColorBrewer_1.1-3          abind_1.4-5                </span></span>
<span><span class="co">##  [37] zlibbioc_1.48.0             Rtsne_0.17                 </span></span>
<span><span class="co">##  [39] GenomicRanges_1.54.1        purrr_1.0.2                </span></span>
<span><span class="co">##  [41] imagerExtra_1.3.2           RCurl_1.98-1.13            </span></span>
<span><span class="co">##  [43] GenomeInfoDbData_1.2.11     IRanges_2.36.0             </span></span>
<span><span class="co">##  [45] S4Vectors_0.40.2            ggrepel_0.9.4              </span></span>
<span><span class="co">##  [47] irlba_2.3.5.1               kohonen_3.0.12             </span></span>
<span><span class="co">##  [49] listenv_0.9.0               spatstat.utils_3.0-4       </span></span>
<span><span class="co">##  [51] goftest_1.2-3               RSpectra_0.16-1            </span></span>
<span><span class="co">##  [53] spatstat.random_3.2-2       fitdistrplus_1.1-11        </span></span>
<span><span class="co">##  [55] parallelly_1.36.0           pkgdown_2.0.7              </span></span>
<span><span class="co">##  [57] leiden_0.4.3.1              codetools_0.2-19           </span></span>
<span><span class="co">##  [59] DelayedArray_0.28.0         RcppRoll_0.3.0             </span></span>
<span><span class="co">##  [61] tidyselect_1.2.0            farver_2.1.1               </span></span>
<span><span class="co">##  [63] imager_0.45.2               matrixStats_1.2.0          </span></span>
<span><span class="co">##  [65] stats4_4.3.2                spatstat.explore_3.2-5     </span></span>
<span><span class="co">##  [67] jsonlite_1.8.8              bmp_0.3                    </span></span>
<span><span class="co">##  [69] ellipsis_0.3.2              progressr_0.14.0           </span></span>
<span><span class="co">##  [71] ggridges_0.5.5              survival_3.5-7             </span></span>
<span><span class="co">##  [73] iterators_1.0.14            systemfonts_1.0.5          </span></span>
<span><span class="co">##  [75] foreach_1.5.2               ggnewscale_0.4.9           </span></span>
<span><span class="co">##  [77] tools_4.3.2                 ragg_1.2.7                 </span></span>
<span><span class="co">##  [79] ica_1.0-3                   Rcpp_1.0.11                </span></span>
<span><span class="co">##  [81] glue_1.6.2                  gridExtra_2.3              </span></span>
<span><span class="co">##  [83] SparseArray_1.2.3           xfun_0.41                  </span></span>
<span><span class="co">##  [85] DESeq2_1.42.0               MatrixGenerics_1.14.0      </span></span>
<span><span class="co">##  [87] GenomeInfoDb_1.38.5         dplyr_1.1.4                </span></span>
<span><span class="co">##  [89] withr_2.5.2                 BiocManager_1.30.22        </span></span>
<span><span class="co">##  [91] fastmap_1.1.1               fansi_1.0.6                </span></span>
<span><span class="co">##  [93] digest_0.6.33               R6_2.5.1                   </span></span>
<span><span class="co">##  [95] mime_0.12                   textshaping_0.3.7          </span></span>
<span><span class="co">##  [97] colorspace_2.1-0            scattermore_1.2            </span></span>
<span><span class="co">##  [99] tensor_1.5                  jpeg_0.1-10                </span></span>
<span><span class="co">## [101] spatstat.data_3.0-3         utf8_1.2.4                 </span></span>
<span><span class="co">## [103] tidyr_1.3.0                 generics_0.1.3             </span></span>
<span><span class="co">## [105] data.table_1.14.10          httr_1.4.7                 </span></span>
<span><span class="co">## [107] htmlwidgets_1.6.4           S4Arrays_1.2.0             </span></span>
<span><span class="co">## [109] uwot_0.1.16                 pkgconfig_2.0.3            </span></span>
<span><span class="co">## [111] gtable_0.3.4                registry_0.5-1             </span></span>
<span><span class="co">## [113] lmtest_0.9-40               XVector_0.42.0             </span></span>
<span><span class="co">## [115] htmltools_0.5.7             carData_3.0-5              </span></span>
<span><span class="co">## [117] pwr_1.3-0                   dotCall64_1.1-1            </span></span>
<span><span class="co">## [119] fftwtools_0.9-11            SeuratObject_5.0.1         </span></span>
<span><span class="co">## [121] scales_1.3.0                png_0.1-8                  </span></span>
<span><span class="co">## [123] tvR_0.3.2                   knitr_1.45                 </span></span>
<span><span class="co">## [125] Signac_1.12.0               reshape2_1.4.4             </span></span>
<span><span class="co">## [127] nlme_3.1-163                cachem_1.0.8               </span></span>
<span><span class="co">## [129] zoo_1.8-12                  stringr_1.5.1              </span></span>
<span><span class="co">## [131] KernSmooth_2.23-22          parallel_4.3.2             </span></span>
<span><span class="co">## [133] miniUI_0.1.1.1              desc_1.4.3                 </span></span>
<span><span class="co">## [135] infix_0.1.0                 pillar_1.9.0               </span></span>
<span><span class="co">## [137] grid_4.3.2                  vctrs_0.6.5                </span></span>
<span><span class="co">## [139] RANN_2.6.1                  ggpubr_0.6.0               </span></span>
<span><span class="co">## [141] promises_1.2.1              car_3.1-2                  </span></span>
<span><span class="co">## [143] xtable_1.8-4                cluster_2.1.4              </span></span>
<span><span class="co">## [145] evaluate_0.23               Rsamtools_2.18.0           </span></span>
<span><span class="co">## [147] cli_3.6.2                   locfit_1.5-9.8             </span></span>
<span><span class="co">## [149] compiler_4.3.2              rlang_1.1.2                </span></span>
<span><span class="co">## [151] crayon_1.5.2                rngtools_1.5.2             </span></span>
<span><span class="co">## [153] ggsignif_0.6.4              future.apply_1.11.1        </span></span>
<span><span class="co">## [155] labeling_0.4.3              plyr_1.8.9                 </span></span>
<span><span class="co">## [157] fs_1.6.3                    stringi_1.8.3              </span></span>
<span><span class="co">## [159] viridisLite_0.4.2           deldir_2.0-2               </span></span>
<span><span class="co">## [161] gridBase_0.4-7              BiocParallel_1.36.0        </span></span>
<span><span class="co">## [163] Biostrings_2.70.1           munsell_0.5.0              </span></span>
<span><span class="co">## [165] tiff_0.1-12                 lazyeval_0.2.2             </span></span>
<span><span class="co">## [167] spatstat.geom_3.2-7         Matrix_1.6-4               </span></span>
<span><span class="co">## [169] RcppHNSW_0.5.0              future_1.33.1              </span></span>
<span><span class="co">## [171] statmod_1.5.0               shiny_1.8.0                </span></span>
<span><span class="co">## [173] highr_0.10                  SummarizedExperiment_1.32.0</span></span>
<span><span class="co">## [175] rbibutils_2.2.16            ROCR_1.0-11                </span></span>
<span><span class="co">## [177] broom_1.0.5                 igraph_1.6.0               </span></span>
<span><span class="co">## [179] memoise_2.0.1               bslib_0.6.1                </span></span>
<span><span class="co">## [181] fastmatch_1.1-4</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Patrick C.N. Martin.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
